src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -95,6 +103,9 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             final Marshaller marshaller = jaxbContext.createMarshaller();
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
+            if(!Files.exists(path.getFileName())) {
```

Comment:
Recall in our discussion that FileWriter API creates the file if the file does not exist. Therefore, placing the check after that call makes it redundant.